### PR TITLE
fix: issue on dependencies auro--select for matchWidth for the value to much long

### LINKED ISSUE #101: Demo for flexible width containers
# General Support Request

With recent updates to the select element using the `matchWidth` attr from auro-dropdown, we should have an example of how this will impact scenarios where the select menu does not appear at full width on the page. 

Screen Shot 2022-09-09 at 2 07 45 PM

## Possible Solution

Screen Shot 2022-09-09 at 2 07 45 PM

```html
<style>
@media (min-width: 900px) {
  .container {
    display: flex;
    justify-content: space-between;
    flex-direction: row;
  }
  .fifty-percent {
    width: 50%;
  }
  .twenty-percent {
    width: 20%;
  }
}

</style>

<div class="container">
  <auro-select placeholder="Wide options" class="twenty-percent">
    <auro-menu>
      <auro-menuoption value="stops">All the stops, so many stops</auro-menuoption>
      <auro-menuoption value="price">Price, price and more about prices</auro-menuoption>
      <auro-menuoption value="duration">Duration from here to there is pretty long</auro-menuoption>
      <auro-menuoption value="departure">Departure time is not for another four hours</auro-menuoption>
      <auro-menuoption value="arrival">Upon arrival we will be greeted with flowers</auro-menuoption>
      <auro-menuoption value="prefer alaska">I really prefer Alaska Airlines for my vacation travels</auro-menuoption>
    </auro-menu>
  </auro-select>

  <auro-select class="fifty-percent" placeholder="Wide selection trigger">
    <auro-menu>
      <auro-menuoption value="stops">Stops</auro-menuoption>
      <auro-menuoption value="price">Price</auro-menuoption>
      <auro-menuoption value="duration">Duration</auro-menuoption>
      <auro-menuoption value="departure">Departure</auro-menuoption>
      <auro-menuoption value="arrival">Arrival</auro-menuoption>
      <auro-menuoption value="prefer alaska">Prefer Alaska</auro-menuoption>
    </auro-menu>
  </auro-select>

  <auro-select placeholder="Default menu">
    <auro-menu>
      <auro-menuoption value="stops">Stops</auro-menuoption>
      <auro-menuoption value="price">Price</auro-menuoption>
      <auro-menuoption value="duration">Duration</auro-menuoption>
      <auro-menuoption value="departure">Departure</auro-menuoption>
      <auro-menuoption value="arrival">Arrival</auro-menuoption>
      <auro-menuoption value="prefer alaska">Prefer Alaska</auro-menuoption>
    </auro-menu>
  </auro-select>
</div>

```

With this demo, there was another issue discovered. If a select menu is shorter than the content of the menu, when a long string is selected, this presents an awkward experience. 

Screen Shot 2022-09-09 at 2 09 55 PM

The element should support truncating the content within the trigger.

### PR DESCRIPTION
**Pull request by irma-kurnia-alaska**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** [Demo for flexible width containers #101](https://github.com/AlaskaAirlines/auro-select/issues/101)

## Summary:

- Set the triggerContentWrapper css to have truncated ui
- set  label css have white-space: nowrap (handle when the label have more text)

I attached the screenshot in my local base the modified:
sfdfd

## Type of change:

Please delete options that are not relevant.

- [ ] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "================================================================================"
   ],
   "name": "@aurodesignsystem/auro-dropdown",
-  "version": "2.10.1",
+  "version": "2.10.2",
```
Comment (fajar-apri-alaska) [inline package.json]: no need to update this manually because Auro use semantic versioning.


edit:
oops, I mean semantic release.

**Review by fajar-apri-alaska — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (irma-kurnia-alaska) [reply package.json]: oh, okay, thanks @fajar-apri-alaska, i will revert this.

**Review by blackfalcon — CHANGES_REQUESTED** [review]
Please add an example of this feature in the demo or apiExamples.

File: scripts/generateDocs.mjs [hunk 2]
```diff
@@ -2,8 +2,9 @@ import path from 'path';
 import markdownMagic from 'markdown-magic';
 import fs from 'fs';
 import https from 'https';
+import { fileURLToPath } from 'url';
 
-const __dirname = new URL('.', import.meta.url).pathname;
+const __dirname = fileURLToPath(new URL('.', import.meta.url));
```
Comment (jason-capsule42) [inline scripts/generateDocs.mjs:7]: This file is coming from a template that all components use and individual components should not diverge.

Is there a case why this change is necessary for this issue?

File: docs/partials/index.md [hunk 3]
```diff
@@ -106,6 +106,22 @@ This common example uses the default `auro-dropdown` element with the attributes
 
 </auro-accordion>
 
+#### Truncated trigger component with fixed width
```
Comment (jason-capsule42) [inline docs/partials/index.md]: Let's move this to docs/partials/api.md and put it with the "Other Examples" section of the page.

It does not belong here as a subjection of this heading.

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]
See comments and also rebase the branch.

File: docs/partials/index.md (hunk 3 above)
Comment (irma-kurnia-alaska) [reply docs/partials/index.md]: ok @jason-capsule42, thanks.

File: scripts/generateDocs.mjs (hunk 2 above)
Comment (fajar-apri-alaska) [reply scripts/generateDocs.mjs:7]: @jason-capsule42, in my case, it doesn't work on my machine.
I have something like `home/fajar/Git Project/auro-dropdown` local path. When I ran the old script, it returns `home/fajar/Git%20Project/auro-dropdown` and somehow it couldn't find the path it refer to.

image

I already asked Dale about this as well, He said it is fine to change that.

Reference: https://github.com/AlaskaAirlines/auro-radio/pull/120/commits/2fcaa20682b749e8bb7fb9340e7f1492db3c19b9

Comment (jason-capsule42) [reply scripts/generateDocs.mjs:7]: Let't not make this change as a one off just to this component repo. @jordanjones243 @blackfalcon what do you think about this proposed change to the script. Is this something you think makes sense to change to all the generateDocs.js files in all components?

**Review by jason-capsule42 — APPROVED** [review]
